mcdreforged/cli/cli_entry.py

```diff
@@ -157,3 +176,121 @@ def write(base_path: str, *, directory_only: bool):
 
 	writeln('Packed {} files/folders into "{}"'.format(file_counter, file_name))
 	writeln('Done')
+
+def init_plugin_workspace(path: str, pid: str, name: str, description: str, authors: str, link: str, resources: list[str], entrypoint: str, archive_name: str, *, quiet: bool = False):
+	writeln = print if not quiet else lambda *args, **kwargs: None
+	if quiet:
+		def ask(*args, default=None, **kwargs):
+			return default
+	else:
+		def ask(msg, *, default=None, skip=False):
+			r = input(('{} (enter to skip): ' if skip else '{}: ' if default is None else '{0} (default "{1}"): ').format(msg, default))
+			return default if len(r) == 0 else r
+
+	if path is None:
+		path = ask('Plugin work directory', default='.')
+
+	metafile = os.path.join(path, plugin_constant.PLUGIN_META_FILE)
+	if os.path.exists(metafile):
+		writeln('[ERROR] Meta file "{}" already exists'.format(metafile))
+		sys.exit(1)
+
+	if pid is None:
+		pid = ask('Id', default=os.path.basename(os.path.abspath(path))).strip()
+	if name is None:
+		name = ask('Name', default=pid).strip()
+	if description is None:
+		description = ask('Description', default='This is a plugin for MCDR').strip()
+	if authors is None:
+		authors = ask("Author(s), split with ':'")
+	if link is None:
+		link = ask('Main page link', skip=True)
+	if resources is None:
+		resources = ask("Resource(s), split with ':'", skip=True)
+	if entrypoint is None:
+		entrypoint = ask('Entry point', skip=True)
+	if archive_name is None:
+		archive_name = ask('Archive name', skip=True)
+
+	metadata = {
+		'id': pid,
+		'version': '1.0.0',
+		'name': name,
+		'description': description,
+		'dependencies': {
+			'mcdreforged': '>=2.0.0'
```

Comment:
Is this really necessary?

Reply:
May not, but many plugins use it

Reply:
It's necessary only if you requires some new features of MCDR I think, so I'll choose to leave it to the user

Reply:
I think it can warn the authors: this plugin can only run with MCDR version >= 2.0.0

Reply:
Nope. It's clear that a .mcdr plugin should only runs on MCDR 2.x so you don't need this requirement.

Reply:
> I think it can warn the authors: this plugin can only run with MCDR version >= 2.0.0

Not necessary since they are already writing this plugin under MCDR 2.x environment and they are not likely to use the plugin in MCDR <2.0

Reply:
I changed it to current mcdreforged version

Reply:
What I mean is, it's **not necessary** to added a MCDR version requirement in the dependencies field

Reply:
If you think that's a better choice. I'll remove it.

Reply:
> It's necessary only if you requires some new features of MCDR I think, so I'll choose to leave it to the user

I have made the choice weeks ago haven't you seen that?

Reply:
I mean, if you insist your choice is the best one, I'll support your choice